example/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
+)
+
+func main() {
+	guest.Filter = api.FilterFunc(nameEqualsPosSpec)
+}
+
+// nameEqualsPosSpec schedules this node if its name equals its pod spec.
```

Comment:
We can come back to such interface topic later, but at final, we probably want to make the interface of wasm guest very similar to the existing scheduling framework interface. 
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/interface.go#L373

Reply:
right I was thinking that `api.Filter` is similar to that, excluding the ctx parameter, and temporarily excluding `state` as I don't have a clear idea what that will map to.

Reply:
Ah, yes. a bit different topic though, we definitely need to support cycle state somehow eventually.
The cycle state is the object which is initialized at the beginning of each scheduling. And we use it as a way to pass something from one extension point to another. One common usecase is pre-calculate something in `PreFilter` or `PreScore` and use pre-calculation result in `Filter` or `Score`. (PreFilter/PreScore is called once per one scheduling, but Filter/Score is called for every potential Nodes. That's why we want to do precalculation instead of calculating every time Filter/Score is called)
https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/#extension-points

Reply:
good. I think we can stub in a state param and develop it further. just like now we stubbed in the other params but they are not fully working yet due to conversion issues

Reply:
I expect that the difficulty to support the cycle state is that people can insert any data in the cycle state.
We don't know what they're, and the cycle state also contains the mutex, which probably difficult to pass.

Probably two options here?
- don't support the cycle state referred from other plugins.
  - = we don't need to pass the cycle state from guest to host; we can just keep it in the guest side.
- require people to define how to serialize their cycle state data. And the guest serialize the data based on that to pass it to host.
  - take a lock via the host function?

Reply:
Anyway created two followup issues to discuss them.

- #15
- #16

Reply:
> we don't need to pass the cycle state from guest to host; we can just keep it in the guest side.

That won't work because currently, different guest instances may be invoked for the same cycle - see https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/pull/6#discussion_r1204239244

Then guest A sets up its cycle state, but guest B gets called later and doesn't have it.

Reply:
Right... But, we may be able to give the same instance for the same Pod's scheduling.
Looking at each scheduling, the plugin is not called in parallel except [during the preemption](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/preemption/preemption.go#LL606C77-L606C77) (I need to have a double check if there are no other places to call same plugin in parallel)

We can take alock or something during the preemption. Yes, then the performance problem may be coming next though.